src/cth_readable_compact_shell.erl

```diff
@@ -51,7 +51,7 @@
 
 -export([terminate/1]).
 
--record(state, {id, suite, groups, opts}).
+-record(state, {id, suite, groups, opts, last_suite, last_tc_failed}).
```

Comment:
Also oof, that gets to be a lot of tracking to know if the last item required a newline.

It might actually be simpler to track, at the end of a piece of output, whether we are on a new line or not, and choose accordingly?

Reply:
I'll change the names; I think your suggestions are sane. 😄 

For your second comment do you mean tracking it via a state flag, too? Or what do you have in mind?

Reply:
Both of the new fields appear to be there specifically to make decisions about whether a newline was output or not. The thing I'm wondering is whether it's worth tracking these semantic things ("we print a new line after a new suite or a failure") vs. tracking plainly "are we on a fresh new line or a line with content" within the state.

I'm unsure either way.

Reply:
I can try that 😄; it's not as simple as I first thought. I'll basically have to redo what I did for these two flags, starting from "no print".

Reply:
Yeah. I don't know if it's even better, and I don't want you to work for nothing, but the sort of oddities baked in the current implementation really go around whether we need a line break or not in a funny way based on settings.

Would it make sense to have a sort of explicit set of "Pre-suite, pre-case, post-case, post-stacktrace" newline set of parameter options picked once at start and then just explicitly going "do I need a line break here or not?" Like the new requirements hint at a need for a restructure given how much repetition and variation is within the individual calls.